homeassistant/components/matter/config_flow.py

```diff
@@ -0,0 +1,69 @@
+"""Config flow for Matter integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from matter_server.client.client import Client
+from matter_server.client.exceptions import CannotConnect, InvalidServerVersion
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_URL
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import aiohttp_client
+
+from .const import DOMAIN, LOGGER
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_URL, default="http://172.30.32.1:5580/chip_ws"): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> None:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+
+    client = Client(data[CONF_URL], aiohttp_client.async_get_clientsession(hass))
+    await client.connect()
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Matter."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        # For now because we use services and they can't distinguish between entries
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            await validate_input(self.hass, user_input)
+        except CannotConnect:
+            errors["base"] = "cannot_connect"
+        except InvalidServerVersion:
+            errors["base"] = "invalid_server_version"
+        except Exception:  # pylint: disable=broad-except
+            LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            return self.async_create_entry(title=user_input[CONF_URL], data=user_input)
```

Comment:
Can't we set a better title than the URL, such as the fabric index/id or just "localhost" is the matter server is local or the IP otherwise.